UPGRADING.md

```diff
@@ -54,4 +64,70 @@ runtime failures of plugins in [OpenSearch plugin Issues](https://github.com/ope
 * `KIB` -> `OSD`
 
 Here is an example for the changes which were done for OpenSearch Dashboard plugin.
-https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/1
\ No newline at end of file
+https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/1
+
+### Backwards Compatibility support
+
+There are a bunch of items to think about and support backwards compatibility.
+
+* Renaming Rest APIs
+* Renaming Indexes
+* Renaming Settings
+* Renaming Namespaces (E.g. com.amazon.opendistroforelasticsearch to org.opensearch)
+* Renaming all remaining identifiers (E.g Opendistro to OpenSearch).
+* 100% Backward compatibility with ES v.?
+* Drop in replacement for Opendistro plugins with OpenSearch plugins
+
+#### OpenSearch Plugins
+
+##### Rest APIs
+
+All APIs which have to be migrated can follow this design.
+From here on the doc will be focussed on an ODFE plugin as an example for the design.
+
+* All plugins extend `BaseRestHandler` class to implement their own RestAPIs.
+* All the existing APIs with `_opendistro` should be supported but for maintenance mode only.
+* All new APIs will be implemented with `_opensearch` (TBD based on [Issue](https://github.com/opensearch-project/opensearch-plugins/issues/13))
+* The method `routes()` is overriden and this should now have both `_opendistro` and `_opensearch` APIs.
+* Add new tests to cover both old and new APIs.
+* Add documentation for the new APIs.
+
+_Example_: Adding an API in Anomaly Detection plugin.
+```java
+public static final String AD_BASE_URI = "/_opendistro/_anomaly_detection";
```

Comment:
Leave just one example of a single URL to avoid confusion.

I believe we said these should start with `LEGACY_` and use some helper to avoid duplicating route code.

Reply:
Sure, i'll update this. We move fast :)